src/components/SummaryCard/SummaryCardContainer/__tests__/SummaryCardContainer.spec.js

```diff
@@ -61,39 +139,171 @@ describe('SummaryCardContainer', () => {
       />
     );
 
-    const { getByTestId } = container;
+    userEvent.tab();
+
+    // The first summary card select:
+    expect(getByLabelText(/test select 0/i)).toHaveFocus();
+
+    userEvent.tab();
 
-    batchActions = getByTestId(batchActionsId);
-    select = getByTestId(`${selectId}${summaryCards[0].id}`);
+    // Loop complete.
+    // The first summary card select:
+    expect(getByLabelText(/test select 1/i)).toHaveFocus();
+
+    userEvent.tab();
   });
 
-  describe('Rendering', () => {
-    it('renders', () => {
-      expect(container.container.firstChild).toMatchSnapshot();
-    });
+  test('should cycle elements in tab order when the batch actions toolbar is activated', () => {
+    const { getByLabelText, getByText } = render(
+      <SummaryCardContainer
+        render={({ getBatchActionProps, getSelectionProps, summaryCards }) => (
+          <Fragment>
+            <SummaryCardBatchActions {...getBatchActionProps()}>
+              <SummaryCardBatchAction
+                tabIndex={getBatchActionProps().shouldShowBatchActions ? 0 : -1}
+              >
+                test batch action
+              </SummaryCardBatchAction>
+            </SummaryCardBatchActions>
+            {summaryCards.map(({ id }) => (
+              <SummaryCard key={id}>
+                <SummaryCardFooter>
+                  <SummaryCardSelect
+                    {...getSelectionProps({ id })}
+                    id={`summary-select--${id}`}
+                    labelText={`test select ${id}`}
+                  />
+                </SummaryCardFooter>
+              </SummaryCard>
+            ))}
+          </Fragment>
+        )}
+        summaryCards={summaryCards}
+      />
+    );
+
+    // Currently there's no way to load the page
+    // with summary cards pre-checked AND batch action toolbar updated.
+    // Therefore, we have to select both cards first:
+    userEvent.click(getByText(/test select 0/i));
+    userEvent.click(getByText(/test select 1/i));
+
+    userEvent.tab();
+
+    // The batch action button:
+    expect(getByText(/test batch action/i)).toHaveFocus();
```

Comment:
Because we leverage the `DataTable`s batch actions here, I'm wondering what your thoughts are on where our responsibilities should end for what we should be testing?

It still feels a bit unclear to me now, but I'd like to understand the boundaries so we're not tightly-coupled to upstream functionality when writing unit tests.

Reply:
For sure -- I'm not necessarily testing the functionality of the data table batch action button in isolation, in this case; I'm testing it here in relation to tab order (and interactivity) within the summary card container. That's not something necessarily controlled in the Carbon's batch action component -- it's moreso how we use it. I think that's a big distinction in this case. 

I think something out of scope would be testing that the component can take in a custom class, or have other typical Carbon button / batch action button props.

Reply:
Great point!